src/view/renderer.js

```diff
@@ -473,19 +536,92 @@ export default class Renderer {
 	 * filler should be rendered.
 	 */
 	_updateChildren( viewElement, options ) {
+		const diff = this._diffElementChildren( viewElement,
+			{ inlineFillerPosition: options.inlineFillerPosition, bind: true, withChildren: true } );
+
+		if ( diff ) {
+			const actions = diff.actions;
+			const domElement = diff.domElement;
+			const actualDomChildren = diff.actualDomChildren;
+			const expectedDomChildren = diff.expectedDomChildren;
+
+			let i = 0;
+			const nodesToUnbind = new Set();
+			for ( const action of actions ) {
+				if ( action === 'insert' ) {
+					insertAt( domElement, i, expectedDomChildren[ i ] );
+					i++;
+				} else if ( action === 'delete' ) {
+					nodesToUnbind.add( actualDomChildren[ i ] );
+					remove( actualDomChildren[ i ] );
+				} else { // 'equal'
+					// Force updating text nodes inside elements which did not change and do not need to be re-rendered (#1125).
+					this._markDescendantTextToSync( this.domConverter.domToView( expectedDomChildren[ i ] ) );
+					i++;
+				}
+			}
+
+			// Unbind removed nodes. When node does not have a parent it means that it was removed from DOM tree during
+			// comparision with the expected DOM. We don't need to check child nodes, because if child node was reinserted,
+			// it was moved to DOM tree out of the removed node.
+			for ( const node of nodesToUnbind ) {
+				if ( !node.parentNode ) {
+					this.domConverter.unbindDomElement( node );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Compares viewElement actual and expected children and actions sequence which can be used to transform
+	 * actual children into expected ones.
+	 *
+	 * @private
+	 * @param viewElement
+	 * @param {Object} options
+	 * @param {module:engine/view/position~Position} options.inlineFillerPosition The position on which the inline
+	 * filler should be rendered.
+	 * @param {Boolean} options.bind If new view elements should be bind to their corresponding DOM elements.
```

Comment:
This is smelly. It violates the single responsibility principle – why does `diffSth()` bind anything?

I noticed this because I couldn't understand the comment above:

> // We do not perform any operations on DOM here so there is no need to bind view element or convert its children.

It didn't have any sense for me in that context because I saw that the line is `const diff = this._diffElementChildren()`.